### PR TITLE
feat: migrates viewing and finding users

### PR DESCRIPTION
**Pull request by tdstein**

### CODE REVIEW

File: 02-users/01-viewing-user-information/index.qmd [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Python
+
+Use the `users` attribute find a user.
+
+```{.python}
+from posit import connect
+
+client = connect.Client()
+user = client.user.find_one()
```
Comment (toph-allen) [inline 02-users/01-viewing-user-information/index.qmd:21]: I think we prob need more detail here. What information can they search for? Just a sentence or two of prose, and an example where they're searching for a specific person, might make it more useful.

File: 02-users/01-viewing-user-information/index.qmd [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "updated_time": "2023-02-17 16:23:09",
+    "active_time": "2023-02-17 16:23:09",
+    "confirmed": true,
+    "locked": false,
+    "guid": "f155520a-ca2e-4084-b0a0-12120b7d1add"
+}
+```
+## R
+
+Use the `get_users` function to find a user
```
Comment (toph-allen) [inline 02-users/01-viewing-user-information/index.qmd:43]:
```suggestion
Use the `get_users` function to return a data frame of all users accounts on the Connect server, which you can filter to find the specific user whose information you need.
```

This function doesn't find a single user, it gets a data frame of all users. We should show the workflow needed to get to the information from a single user (i.e. just a filter step in dplyr).

File: 02-users/02-finding-users/index.qmd [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+:::
+
+## Discussion
+
+These methods provide an effective way to search for users when you do not know the users unique identifier (guid).
+
+There are additional parameter that you can use to filter the resulting users set.
+
+You can use the `prefix` parameter to find users by username, first name, or last name.
```
Comment (toph-allen) [inline 02-users/02-finding-users/index.qmd:80]: My gut impulse is that this sort of information should go under the "solution", as it's kind of core information about the workflows.

File: 02-users/02-finding-users/index.qmd [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+You want to find users.
+
+## Solution
+
+::: {.panel-tabset group="language"}
+
+## Python
+
+Use the `users.find` and `users.find_one` methods to find users using various criteria.
```
Comment (toph-allen) [inline 02-users/02-finding-users/index.qmd:15]: What criteria can we search on? I think this is an important enough part of the solution to include in the prose, and to show an example of in the code.

**Review by toph-allen — COMMENTED** [review]
I feel like my impulse is almost always going to be towards slightly more verbosity and explanation in recipes. 😂

I kinda feel like this doesn't need to be two separate recipes, especially since "finding users" in Python also explains `find_one()`, and the R example is the same for each. It's like, from the users' point of view, there isn't really much difference between finding all users and finding a single one, and especially if the args are the same in Python for each, we could just explain `find_one()` alongside `find()`.

**Comment by tdstein** [comment]
> I feel like my impulse is almost always going to be towards slightly more verbosity and explanation in recipes. 😂
> 
> I kinda feel like this doesn't need to be two separate recipes, especially since "finding users" in Python also explains `find_one()`, and the R example is the same for each. It's like, from the users' point of view, there isn't really much difference between finding all users and finding a single one, and especially if the args are the same in Python for each, we could just explain `find_one()` alongside `find()`.

I've been separating it by "how do I view user information" and "how to do I search for a user". I agree there is a bit of overlap, but I think it's two common questions that happen to point toward the same solution.

File: 02-users/02-finding-users/index.qmd (hunk 3 above)
Comment (tdstein) [reply 02-users/02-finding-users/index.qmd:80]: Yeah, I've been leaning on examples like these: https://rc2e.com/datastructures#recipe-id062

Which seems to keep the solution short and sweet. Then go into more detail about using the function in the discussion.

Comment (toph-allen) [reply 02-users/02-finding-users/index.qmd:66]: Cool! I'm down with that approach.

File: 02-users/02-finding-users/index.qmd [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+2 hadley@posit.co     hadley     Hadley     Wickham   publisher 2015-05-20 19:44:37
+# ℹ 5 more variables: updated_time <dttm>, active_time <dttm>, confirmed <lgl>,
+#   locked
+```
+
+:::
+
+## Discussion
+
+These methods provide an effective way to search for users when you do not know the users unique identifier (guid).
```
Comment (toph-allen) [inline 02-users/02-finding-users/index.qmd:62]: This sentence kinda implies that you can search for users by GUID, but I don't see that mentioned elsewhere in these articles. Either should remove it from here or give an example of it.

File: 02-users/02-finding-users/index.qmd [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+There are additional parameter that you can use to filter the resulting users set.
+
+You can use the `prefix` parameter to find users by username, first name, or last name.
```
Comment (toph-allen) [inline 02-users/02-finding-users/index.qmd:66]:
```suggestion
You can use the `prefix` parameter to search for users with a username, first name, or last name matching the provided string.
```

**Review by toph-allen — APPROVED** [review]
Still some comments, but you can fix them here or not, and merge, and we'll fix forwards.

**Pull request merged**